controllers/serviceaccount_controller.go

```diff
@@ -89,9 +89,10 @@ func AddServiceAccountProviderControllerToManager(ctx *context.ControllerManager
 			&source.Kind{Type: &vmwarev1.ProviderServiceAccount{}},
 			handler.EnqueueRequestsFromMapFunc(r.providerServiceAccountToVSphereCluster),
 		).
+		// Watches the secrets to re-enqueue once the service account token is set
 		Watches(
-			&source.Kind{Type: &corev1.ServiceAccount{}},
-			handler.EnqueueRequestsFromMapFunc(r.serviceAccountToVSphereCluster),
+			&source.Kind{Type: &corev1.Secret{}},
```

Comment:
why are we watching secrets instead of SAs? sure we're creating the secret now, but that's still done because there's a new service account being setup.

Reply:
Because the generated secret eventually gets populated with the token for the service account, which needs to be synced over to the guest cluster